src/silo/zstdfasta/zstdfasta_table.cpp

```diff
@@ -0,0 +1,82 @@
+#include "silo/zstdfasta/zstdfasta_table.h"
+
+#include <fmt/format.h>
+#include <duckdb.hpp>
+
+#include "silo/common/fasta_reader.h"
+#include "silo/preprocessing/preprocessing_exception.h"
+#include "silo/zstdfasta/zstd_compressor.h"
+#include "silo/zstdfasta/zstdfasta_reader.h"
+
+namespace {
+void initializeTable(duckdb::Connection& connection, std::string table_name) {
+   auto return_value = connection.Query(fmt::format("DROP TABLE IF EXISTS {};", table_name));
+   if (return_value->HasError()) {
+      throw silo::preprocessing::PreprocessingException(return_value->ToString());
+   }
+   return_value = connection.Query(fmt::format(
+      "CREATE TABLE {} ("
+      "    key STRING,"
+      "    sequence BLOB"
+      ");",
+      table_name
+   ));
+   if (return_value->HasError()) {
+      throw silo::preprocessing::PreprocessingException(return_value->ToString());
+   }
+}
+}  // namespace
+
+void silo::ZstdFastaTable::generate(
```

Comment:
Do we need this function? My IDE only find usages in tests.

Reply:
Currently not, but it will be needed to allow more flexible inputs. And the more obvious testing is a great pro as well

Reply:
At least in the current implementation those are two distinct function. Can we at least extract some common parts?